group-generator.js

```diff
@@ -18,53 +18,49 @@ let period1 = [
   {name: 'Gollum', level: 2}
 ];
 
-
 function addRandomSortNumber(array) {
-  array.map(function(student){
-  student.sortNumber = Math.random();
-
+  array.map(function (student) {
```

Comment:
This is a Doug Crockford recommendation that I really like. Whenever you have an anonymous function, keep the arguments parens away from the word `function` in order to quickly show the viewer that it is an un-named function. E.g. `function (foo) {...}` or `function bar(foo) {...}` if the function is named.

You'll see this a few times throughout the PR.